draft-thomson-http-oblivious.md

```diff
@@ -210,56 +210,69 @@ Responses are bound to responses and so consist only of AEAD-protected content.
 {{response}} describes the process for constructing and processing an
 Encapsulated Response.
 
+~~~
+Encapsulated Response {
+  Nonce (Nk),
+  AEAD-Protected Response (..),
+}
+~~~
+{: #fig-enc-response title="Encapsulated Response"}
+
+The size of the Nonce field in an Encapsulated Response corresponds to the
+size of an AEAD key for the corresponding HPKE ciphersuite.
+
 ## HPKE Encapsulation of Requests {#request}
 
 Clients encapsulate a request `request` with an HPKE public key `pkR`,
-whose Key Identifier is `keyID` as follows:
+whose wire-encoded Key Identifier is `keyID` as follows:
 
 1. Compute an HPKE context using `pkR`, yielding `context` and encapsulation
    key `enc`.
 
 2. Construct additional associated data, `aad`, by prepending a single byte
-   with a value of 0x01 to the key identifier. The key identifier length is not
+   with a value of 0x01 to the key identifier. The key identifier length is
```

Comment:
So I'm not so sure about this one.  One problem is that the varint encoding is only injective, not bijective.  That means that you can varint encode the length differently for encoding and the AAD if you aren't careful to also keep the length of the length.  That breaks things if implemented as described.

My understanding was that an AEAD was required to authenticate the length of the AAD in addition to its content.  GCM does: `S = GHASH_H (A || 0^v || C || 0^u || [len(A)]_64 || [len(C)]_64).`  Can't we rely on that?

Reply:
Maybe? But including the length is cheap, so why *not* do it?

Reply:
I just explained why: it makes it difficult to implement.

Reply:
Yeah, I hear you, I'm just worried about what might happen if you don't authenticate the length explicitly. (I don't see any immediate problem, but I'd prefer we just explicitly authenticate every bit sent on the wire. I'm fine moving this to a separate issue.)

Reply:
If this were input to key schedule, sure.  But I'm not even sure we need to include the keyID in AAD at all.  It's convenient, so sure, but it isn't *necessary*.

Reply:
Reverted -- we can discussed separately in #12.